docs/development/modules.md

```diff
@@ -46,6 +46,7 @@ Druid's extensions leverage Guice in order to add things at runtime.  Basically,
 1. Add new Jersey resources by calling `Jerseys.addResource(binder, clazz)`.
 1. Add new Jetty filters by extending `org.apache.druid.server.initialization.jetty.ServletFilterHolder`.
 1. Add new secret providers by extending `org.apache.druid.metadata.PasswordProvider`.
+1. Add new ingest transform by implementing the `org.apache.druid.segment.transform.Transform` interface from the `druid-processing` package.
```

Comment:
Can you add a section in here describing how to register the custom transformation similar to what's documented for adding support for an Input Source (https://github.com/apache/druid/blob/118e89c1e2ae9ea0e1c923c355bff6f1ccacf756/docs/development/modules.md#adding-support-for-a-new-input-source) or a Password Provider (https://github.com/apache/druid/blob/118e89c1e2ae9ea0e1c923c355bff6f1ccacf756/docs/development/modules.md#adding-a-new-password-provider-implementation)

Reply:
> It makes sense to me to add a new item in the list of 15 items under "Writing your own extensions" for completeness sake. I'd be happy to write a dedicated section of these docs, although I think this page seems destined to be split into many pages: one describing the approach for each kind of extension. 

Hi @mitchlloyd, I assume you are interested in writing docs what @suneet-s mentioned based on your comment above. Do you want to add them in this PR? I think that could be done in a follow-up PR if you want.